### ISSUE TITLE
App crashes when using empty Observable arrays

### ISSUE THREAD

**Issue by garie** [ReactiveKit/Bond#341]
I have noticed this in a couple of places but there may be more. This has led to some pretty ugly code where I have to check isEmpty before doing each of these. I believe this happens with types other than `MutableObservable2DArray` but don't have any examples right now.

Code examples:

forEach over empty array
```
let array = MutableObservable2DArray<Bool, Bool>([])
array.forEach { a in
    print(a)
}
```

appendSection to empty array
```
let section = Observable2DArraySection<Bool, Bool>(metadata: true, items: [true])
let array = MutableObservable2DArray<Bool, Bool>([])
array.appendSection(section)
```

index into empty array
```
let array = MutableObservable2DArray<Bool, Bool>([])
let index = array.index { item in
    return true
}
```

Also, this causes my app to segfault on compilation:

```
let array = MutableObservable2DArray<Bool, Bool>()
```

**Comment by srdanrasic** [comment]
Thanks for reporting this @garie! I've just released v5.3.1 with the fix. I think the issue is limited to `(Mutable)Observable2DArray` due to how indices are calculated.

**Comment by garie** [comment]
Thanks! I updated and was able to remove my workarounds. I'll file another if I see similar issues with other arrays in the future.
